examples/AvroGeneric/Program.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using Avro;
+using Avro.Generic;
+using Confluent.Kafka.SchemaRegistry;
+using Confluent.Kafka.SchemaRegistry.Serialization;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+
+namespace Confluent.Kafka.Examples.AvroGeneric
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string brokerList = "localhost:9092";
+            string topicName = "testAvro";
+
+            string userSchema = File.ReadAllText("User.asvc");
+
+            var schemaRegistry = new Mock<ISchemaRegistryClient>();
+            schemaRegistry.Setup(x => x.GetRegistrySubject(It.IsAny<string>(), It.IsAny<bool>())).Returns((string topic, bool isKey) => topic + (isKey ? "-key" : "-value"));
+
+            schemaRegistry.Setup(x => x.RegisterAsync(topicName + "-key", It.IsAny<string>())).ReturnsAsync(1);
+            schemaRegistry.Setup(x => x.RegisterAsync(topicName + "-value", It.IsAny<string>())).ReturnsAsync(2);
+
+            schemaRegistry.Setup(x => x.GetSchemaAsync(1)).ReturnsAsync("string");
+            schemaRegistry.Setup(x => x.GetSchemaAsync(2)).ReturnsAsync(User._SCHEMA.ToString());
+
+            var producerConfig = new Dictionary<string, object> { { "bootstrap.servers", brokerList } };
+            var consumerConfig = new Dictionary<string, object> { { "bootstrap.servers", brokerList }, { "group.id", Guid.NewGuid() } };
+            var consumerDynamicConfig = new Dictionary<string, object> { { "bootstrap.servers", brokerList }, { "group.id", Guid.NewGuid() } };
+
+            var deserializer = new AvroDeserializer(schemaRegistry.Object);
+
+            using (var consumer = new Consumer<object, object>(consumerConfig, deserializer, deserializer))
+            using (var consumerDynamic = new Consumer<dynamic, dynamic>(consumerDynamicConfig, deserializer, deserializer))
```

Comment:
I don't understand how dynamic helps here? The class delivered by the consumer is always the same right? (unless you were thinking of extending things to make some fancy anonymous class instances, but I don't think that's a great idea).

Reply:
The way on avro side for generics (well, it returns an object, so dynamic seems ok). I don't really like dynamic, but it seemed ok here.